### ISSUE TITLE
Unclosed arrays can cause problems for On Demand

### ISSUE THREAD

**Issue by jkeiser** [simdjson/simdjson#1210]
JSON with unclosed arrays end up passing the user data from padding. Actually parsing that data could cause overruns.

Consider this program:

```c++
auto json = "[ 1, "_padded;
for (uint64_t val : parser.parse()) { cout << num; }
```

When the parser gets past the comma, it passes the next structural to the number parser. There is no next structural, however. In the case above, it just fails with NUMBER_ERROR, which is about what you would expect, but if the remaining padding had 16 digits in it, an overrun could occur.

This is an unlikely problem. It also occurred with the DOM version of simdjson, and wasn't discovered until a code inspection years later. Nonetheless, it is a problem!

**Comment by lemire** [comment]
How do we characterize it. Can we document the issue and still release, or do we need to fix it before we release?

**Comment by jkeiser** [comment]
We may want to characterize On Demand as a preview feature in this release, so we can collect feedback and feel free to change the API in the next release if we need to. If we did that, we could definitely release with this in it.

Even if we didn't do that, we could likely document and release in the "very low likelihood combination" category.

**Comment by lemire** [comment]
Yes, I like the preview feature idea. We still should try to explain as carefully as possible how to avoid the problem. Padding with spaces or nulls would be safe, right?

**Comment by jkeiser** [comment]
Yep! If the padding has at least one space or null in it, it is safe, I believe.

**Comment by jkeiser** [comment]
This was fixed with #1453.
